js/core/utils/data.js

```diff
@@ -191,7 +191,13 @@ export const toComparable = function(value, caseSensitive) {
     }
 
     if(!caseSensitive && typeof value === 'string') {
-        return value.toLowerCase();
+        if(options?.collatorOptions?.sensitivity === 'base') {
```

Comment:
Minor comment

Let's remove the first redundant ```?```
(because options always defined and equals ```{}``` by default)

```suggestion
        if(options.collatorOptions?.sensitivity === 'base') {
```